graphql/query_resolver.go

```diff
@@ -963,6 +963,61 @@ func (r *queryResolver) TaskNamesForBuildVariant(ctx context.Context, projectIde
 	return buildVariantTasks, nil
 }
 
+// Waterfall is the resolver for the waterfall field.
+func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions) (*Waterfall, error) {
+	projectId, err := model.GetIdForProject(options.ProjectIdentifier)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("finding project with id: '%s'", options.ProjectIdentifier))
+	}
+	limit := model.DefaultWaterfallVersionLimit
+	if utility.FromIntPtr(options.Limit) != 0 {
+		limit = utility.FromIntPtr(options.Limit)
```

Comment:
Would it be worthwhile to put a hard limit on this? If people can craft their own GQL queries, could they manually request something expensive, like the entire waterfall history?